nixos/modules/services/cluster/hadoop/hbase.nix

```
@@ -5,11 +5,89 @@ let
   cfg = config.services.hadoop;
   hadoopConf = "${import ./conf.nix { inherit cfg pkgs lib; }}/";
   mkIfNotNull = x: mkIf (x != null) x;
+  # generic hbase role options
+  hbaseRoleOption = name: extraOpts: {
+    enable = mkEnableOption (mdDoc "HBase ${name}");
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc "Open firewall ports for HBase ${name}.";
+    };
+
+    restartIfChanged = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc "Restart ${name} con config change.";
+    };
+
+    extraFlags = mkOption {
+      type = with types; listOf str;
+      default = [];
+      description = mdDoc "Extra flags for the ${name} service.";
```

Adding an example to the non-boolean options might be helpful for someone new to HBase.
```suggestion
      description = mdDoc "Extra flags for the ${name} service.";
      example = [ "-XX:+CMSParallelRemarkEnabled" ];
```

---

I'm not sure you can directly pass JVM flags as command line args. I'll add an example HBase flag.